### PR TITLE
4652 translate is immunisation program

### LINKED ISSUE #4652: Mark programs as Immunisation programs
Currently immunisation programs are a separate kind of program, only stored in OMS Central. We need to change this to creating a program in legacy mSupply, then marking that program as an immunisation in Open mSupply. This will allow us to use existing patient program functionality.

STEPS:
- [x] mSupply change - move is_immunisation flag to mSupply
- [x] Update translation to map in this new flag
- [x] bump sync versions
- [x] UI changes - remove Immunisation program create button - list should already contain all immunisation programs
- [x] Remove delete immunisation programs... they would be deleted in mSupply - ensure this is a soft delete though? if there is patient data they need to be soft deleted
- [ ] Remove immunisation program context?

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #4652

# 👩🏻‍💻 What does this PR do?

Translates is_immunisation from mSupply 

You will need this mSupply branch: https://github.com/msupply-foundation/msupply/pull/15168

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Per epic, set up at least one program as a master list
- [ ] Sync
- [ ] It shows in OMS central server in programs > immunisations 

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: server/service/src/sync/translations/program_requisition_settings.rs [hunk 1]
```diff
@@ -180,7 +181,7 @@ fn generate_requisition_program(
         master_list_id: Some(master_list.id.clone()),
         name: master_list.description.clone(),
         context_id: context_row.id.clone(),
-        is_immunisation: false,
+        is_immunisation: master_list.is_immunisation.clone(),
```
Comment (Chris-Petty) [inline server/service/src/sync/translations/program_requisition_settings.rs:184]: nit: bool doesn't need cloning

**Review by Chris-Petty — CHANGES_REQUESTED** [review]
Not sure why but here test error:

```
failures:

---- sync::test::pull_and_push::test_sync_pull_and_push stdout ----
thread 'sync::test::pull_and_push::test_sync_pull_and_push' panicked at repository/src/db_diesel/master_list_name_join.rs:92:9:
assertion `left == right` failed
  left: Ok(None)
 right: Ok(Some(MasterListNameJoinRow { id: "A7A06D78361041448B836857ED4330C4", master_list_id: "87027C44835B48E6989376F42A58F7E3", name_link_id: "name_store_a" }))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Fine on develop though!

**Comment by lache-melvin** [comment]
> Not sure why but here test error:

Ah great catch thanks Chris, was only running the program tests 🙈 Forgot to update the other master list translation test data! 🙏

**Review by Chris-Petty — APPROVED** [review]
beaut

**Pull request merged**